samples/radio_test/src/radio_test.c

```diff
@@ -0,0 +1,565 @@
+				NRF_RADIO_SHORT_READY_START_MASK |
+				NRF_RADIO_SHORT_END_DISABLE_MASK);
+	nrf_radio_txpower_set(NRF_RADIO, txpower);
+	radio_channel_set(mode, channel);
+
+	/* We let the TIMER start the radio transmission again. */
+	nrfx_timer_disable(&timer);
+	nrf_timer_shorts_disable(timer.p_reg, ~0);
+	nrf_timer_int_disable(timer.p_reg, ~0);
+
+	nrfx_timer_extended_compare(&timer,
+		NRF_TIMER_CC_CHANNEL1,
+		nrfx_timer_us_to_ticks(&timer, delay_time),
+		NRF_TIMER_SHORT_COMPARE1_CLEAR_MASK,
+		true);
+
+	nrfx_timer_clear(&timer);
+	nrfx_timer_enable(&timer);
+}
+
+void radio_test_start(const struct radio_test_config *config)
+{
+	switch (config->type) {
+	case UNMODULATED_TX:
+		radio_unmodulated_tx_carrier(config->mode,
+			config->params.unmodulated_tx.txpower,
+			config->params.unmodulated_tx.channel);
+		break;
+	case MODULATED_TX:
+		radio_modulated_tx_carrier(config->mode,
+			config->params.modulated_tx.txpower,
+			config->params.modulated_tx.channel,
+			config->params.modulated_tx.pattern);
+		break;
+	case RX:
+		radio_rx(config->mode,
+			config->params.rx.channel,
+			config->params.rx.pattern);
+		break;
+	case TX_SWEEP:
+		radio_sweep_start(config->params.tx_sweep.channel_start,
+			config->params.tx_sweep.delay_ms);
+		break;
+	case RX_SWEEP:
+		radio_sweep_start(config->params.rx_sweep.channel_start,
+			config->params.rx_sweep.delay_ms);
+		break;
+	case MODULATED_TX_DUTY_CYCLE:
+		radio_modulated_tx_carrier_duty_cycle(config->mode,
+			config->params.modulated_tx_duty_cycle.txpower,
+			config->params.modulated_tx_duty_cycle.channel,
+			config->params.modulated_tx_duty_cycle.pattern,
+			config->params.modulated_tx_duty_cycle.duty_cycle);
+		break;
+	}
+}
+
+void radio_test_cancel(void)
+{
+	nrfx_timer_disable(&timer);
+	radio_disable();
+}
+
+void radio_rx_stats_get(struct radio_rx_stats *rx_stats)
+{
+	size_t size;
+
+#if USE_MORE_RADIO_MODES
+	nrf_radio_mode_t radio_mode;
+
+	radio_mode = nrf_radio_mode_get(NRF_RADIO);
+	if (radio_mode == NRF_RADIO_MODE_IEEE802154_250KBIT) {
+		size = IEEE_MAX_PAYLOAD_LEN;
+	} else {
+		size = sizeof(rx_packet);
+	}
+#else
+	size = sizeof(rx_packet);
+#endif /* USE_MORE_RADIO_MODES */
+
+	rx_stats->last_packet.buf = rx_packet;
+	rx_stats->last_packet.len = size;
+	rx_stats->packet_cnt = rx_packet_cnt;
+}
+
+#if NRF_POWER_HAS_DCDCEN_VDDH || NRF_POWER_HAS_DCDCEN
+void toggle_dcdc_state(u8_t dcdc_state)
+{
+	bool is_enabled;
+
+#if NRF_POWER_HAS_DCDCEN_VDDH
+	if (dcdc_state == 0) {
+		is_enabled = nrf_power_dcdcen_vddh_get(NRF_POWER);
+		nrf_power_dcdcen_vddh_set(NRF_POWER, !is_enabled);
+		return;
+	}
+#endif /* NRF_POWER_HAS_DCDCEN_VDDH */
+
+#if NRF_POWER_HAS_DCDCEN
+	if (dcdc_state <= 1) {
+		is_enabled = nrf_power_dcdcen_get(NRF_POWER);
+		nrf_power_dcdcen_set(NRF_POWER, !is_enabled);
+		return;
+	}
+#endif /* NRF_POWER_HAS_DCDCEN */
+}
+#endif /* NRF_POWER_HAS_DCDCEN_VDDH || NRF_POWER_HAS_DCDCEN */
+
+static void timer_handler(nrf_timer_event_t event_type, void *context)
```

Comment:
I would prefer to use nrfx drivers so that peripheral-related code in ```radio_test``` stays at the same level of abstraction. There is no Radio shim in Zephyr so it is not possible to migrate this code fully to Zephyr driver APIs.